src/ts/background/handlers/Extension.ts

```diff
@@ -50,69 +56,89 @@ export default class Extension {
     return true
   }
 
-  private authorizeApprove ({ id }: MessageAuthorizeApprove): boolean {
+  private authorizeApprove ({ id }: RequestAuthorizeApprove): boolean {
     const queued = this.state.getAuthRequest(id)
     assert(queued, 'Unable to find request')
     const { resolve } = queued
     resolve(true)
     return true
   }
 
-  private authorizeReject ({ id }: MessageAuthorizeReject): boolean {
+  private authorizeReject ({ id }: RequestAuthorizeReject): boolean {
     const queued = this.state.getAuthRequest(id)
     assert(queued, 'Unable to find request')
     const { reject } = queued
     reject(new Error('Rejected'))
     return true
   }
 
-  private signingApprove ({ id, password }: MessageExtrinsicSignApprove): boolean {
+  private signingApprovePassword ({ id, password }: RequestSigningApprovePassword): boolean {
     const queued = this.state.getSignRequest(id)
     assert(queued, 'Unable to find request')
     const { request, resolve, reject } = queued
-    if (!keyringVault.accountExists(request.address)) {
+    if (!keyringVault.accountExists(request.inner.address)) {
       reject(new Error('Unable to find account'))
       return false
     }
-    const payload = createType('ExtrinsicPayload', request, { version: request.version })
     keyringVault.unlock(password).then(() => {
-      const pair = keyringVault.getPair(request.address)
-      const signature = payload.sign(pair)
+      const inner = request.inner
+      const pair = keyringVault.getPair(inner.address)
+      let registry
+      if ((inner as SignerPayloadRaw).data) {
+        registry = new TypeRegistry()
+      } else {
+        const signerPayload = (inner as SignerPayloadJSON)
+        const network = findNetwork(signerPayload.genesisHash)
+        registry = network.registry
+      }
+      const result = request.sign(registry, pair)
```

Comment:
should we define two sign method instead?  one take registry and pair and the other only take pair.

Reply:
Good point. I just followed Jaco's implementation. Probably we can ask Jaco. I think a better interface could be `sign (pair, registry?)` in `RequestSign`. For now, I think let's just keep it more synced with Jaco's implementation, so that future sync is easier.